x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationServiceIntegTests.java

```diff
@@ -121,6 +125,45 @@ public void testGetRoleDescriptorsIntersectionForRemoteCluster() throws IOExcept
         );
     }
 
+    public void testCrossClusterAccessWithInvalidRoleDescriptors() {
+        final String nodeName = internalCluster().getRandomNodeName();
+        final ThreadContext threadContext = internalCluster().getInstance(SecurityContext.class, nodeName).getThreadContext();
+        final AuthorizationService authzService = internalCluster().getInstance(AuthorizationService.class, nodeName);
+        final CrossClusterAccessSubjectInfo crossClusterAccessSubjectInfo = AuthenticationTestHelper.randomCrossClusterAccessSubjectInfo(
+            new RoleDescriptorsIntersection(
+                randomValueOtherThanMany(
+                    rd -> false == (rd.hasClusterPrivileges()
+                        || rd.hasApplicationPrivileges()
+                        || rd.hasConfigurableClusterPrivileges()
+                        || rd.hasRunAs()
+                        || rd.hasRemoteIndicesPrivileges()),
```

Comment:
Nit: can we reuse `RoleDescriptorStore.hasPrivilegesOtherThanIndex` here?

Reply:
Yup. I want to make sure we're not relying on code to test itself though (currently nothing tests `hasPrivilegesOtherThanIndex`, other than the code in `AuthorizationServiceIntegTests`). To address this, I will:

1. Move `hasPrivilegesOtherThanIndex` directly to RoleDescriptor
2. Add a unit test for it
3. Re-use it here